superset/row_level_security/commands/update.py

```diff
@@ -0,0 +1,60 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+
+import logging
+from typing import Any, Dict, Optional
+
+from superset.commands.base import BaseCommand
+from superset.connectors.sqla.models import RowLevelSecurityFilter, SqlaTable
+from superset.dao.exceptions import DAOCreateFailedError
+from superset.extensions import appbuilder, db, security_manager
+from superset.row_level_security.commands.exceptions import RLSRuleNotFoundError
+from superset.row_level_security.dao import RLSDAO
+
+logger = logging.getLogger(__name__)
+
+
+class UpdateRLSRuleCommand(BaseCommand):
+    def __init__(self, model_id: int, data: Dict[str, Any]):
+        self._model_id = model_id
+        self._properties = data.copy()
+        self._model: Optional[RowLevelSecurityFilter] = None
+
+    def run(self) -> Any:
+        self.validate()
+        try:
+            rule = RLSDAO.update(self._model, self._properties)
+        except DAOCreateFailedError as ex:
+            logger.exception(ex.exception)
+            raise DAOCreateFailedError
+
+        return rule
+
+    def validate(self) -> None:
+        self._model = RLSDAO.find_by_id(self._model_id)
+        if not self._model:
+            raise RLSRuleNotFoundError()
+
+        roles = security_manager.find_roles_by_id(self._properties.get("roles", []))
```

Comment:
better to use `populate_roles` from https://github.com/apache/superset/blob/master/superset/commands/utils.py#L66
free validation and raises if a role does not exist

Reply:
done.